### PR TITLE
[db-up] Upgrade pymssql to support Python 3.10

### PR DESCRIPTION
**Pull request by evelyn-ys**
Fix https://github.com/Azure/azure-cli-extensions/issues/4608

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

File: src/db-up/setup.py [hunk 1]
```diff
@@ -40,7 +40,7 @@
     license='MIT',
     author='Microsoft Corporation',
     author_email='azpycli@microsoft.com',
-    url='https://github.com/Azure/azure-cli-extensions/tree/main/src/db-up',
+    url='https://github.com/Azure/azure-cli-extensions/tree/master/src/db-up',
```
Comment (evelyn-ys) [inline src/db-up/setup.py:43]:
```suggestion
    url='https://github.com/Azure/azure-cli-extensions/tree/main/src/db-up',
```

**Comment by yonzhan** [comment]
db-up

File: src/db-up/azext_db_up/custom.py [hunk 2]
```diff
@@ -209,7 +209,7 @@ def _ensure_pymssql():
         python_path = os.environ.get('PYTHONPATH', '')
         os.environ['PYTHONPATH'] = python_path + ':' + db_up_ext_path if python_path else db_up_ext_path
         cmd = [sys.executable, '-m', 'pip', 'install', '--target', db_up_ext_path,
-               'pymssql==2.2.2', '-vv', '--disable-pip-version-check', '--no-cache-dir']
+               'pymssql', '-vv', '--disable-pip-version-check', '--no-cache-dir']
```
Comment (jiasli) [inline src/db-up/azext_db_up/custom.py:212]: This this safe? 🤔

Comment (evelyn-ys) [reply src/db-up/azext_db_up/custom.py:212]: I'm also hesitant about this. But it's annoying to bump manually each time

Comment (jiasli) [reply src/db-up/azext_db_up/custom.py:212]: We are already manually bumming version in `setup.py`, right? If we want, we can loosen the version in both places.

Comment (evelyn-ys) [inline src/db-up/azext_db_up/custom.py:212]:
```suggestion
               'pymssql~=2.2.4', '-vv', '--disable-pip-version-check', '--no-cache-dir']
```

File: src/db-up/setup.py [hunk 3]
```diff
@@ -30,7 +30,7 @@
 ]
 
 if sys.platform != 'darwin':
-    DEPENDENCIES.append('pymssql==2.2.2')
+    DEPENDENCIES.append('pymssql==2.2.4')
```
Comment (evelyn-ys) [inline src/db-up/setup.py:33]:
```suggestion
    DEPENDENCIES.append('pymssql~=2.2.4')
```

**Review by evelyn-ys — COMMENTED** [review]
Use `~=` instead of `==`

**Comment by jiasli** [comment]
`~=` means to loosen the last version. https://peps.python.org/pep-0440/#compatible-release

**Review by jiasli — APPROVED** [review]

**Pull request merged**